src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporterOptions.cs

```diff
@@ -183,6 +183,16 @@ public Uri Endpoint
         /// </remarks>
         public Func<HttpClient> HttpClientFactory { get; set; }
 
+        /// <summary>
+        /// Gets or sets the directory for offline storage.
+        /// </summary>
+        public string StorageDirectory { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether offline storage should be used.
+        /// </summary>
+        public bool EnableOfflineStorage { get; set; }
```

Comment:
Previously, we somewhat decided the OTLP options class would be limited to options defined in the spec. I say somewhat because we haven't stayed 100% true to this with the following:

1) `HttpClientFactory`. Personally, I supported this one because exposing complete control over the transport client of an exporter seems reasonable to me.
2) `ExportProcessorType` and `BatchExportProcessorOptions`. In the past we've discussed deprecating these in favor of a separate processor options class.

Few questions:

1. Do we expect these options to be proposed with the spec?
2. Do we still feel that this options class should be somewhat constrained to the spec?
3. Would these new options be better suited in a separate options class?

Reply:
One of the other nice things https://github.com/open-telemetry/opentelemetry-dotnet/pull/4121#discussion_r1092313085 would do is allow us to drop all options completely. We could have a property on options for setting the factory but I feel like there are ways we could avoid that too.

Reply:
Not including any new propertied to this options class now. The options will all be given at the time of service configuration.